packages/nextjs/src/app-router/client/useInternalNavFun.ts

```diff
@@ -6,31 +6,48 @@ import type { NextClerkProviderProps } from '../../types';
 
 declare global {
   interface Window {
-    __clerk_internal_navFun: NonNullable<
-      NextClerkProviderProps['routerPush'] | NextClerkProviderProps['routerReplace']
+    __clerk_internal_navigations: Record<
+      string,
+      {
+        fun: NonNullable<NextClerkProviderProps['routerPush'] | NextClerkProviderProps['routerReplace']>;
+        promisesBuffer: Array<() => void> | undefined;
+      }
     >;
-    __clerk_internal_navPromisesBuffer: Array<() => void> | undefined;
   }
 }
 
+const registerNavigationType = (name: string) => {
+  if (!window.__clerk_internal_navigations) {
+    window.__clerk_internal_navigations = {};
+  }
+
+  if (!(name in window.__clerk_internal_navigations)) {
+    // @ts-ignore
+    window.__clerk_internal_navigations[name] = {};
+  }
+
+  return window.__clerk_internal_navigations[name];
+};
```

Comment:
nit, this feels more like a getter:

```suggestion
const getClerkNavigationObject = (name: string) => {
    window.__clerk_internal_navigations ??= {};
    window.__clerk_internal_navigations[name] ??= {}
  	return window.__clerk_internal_navigations[name];
};
```